modules/diary.py

```diff
@@ -237,7 +237,63 @@ def complete_task():
         chalk.red(
             'There are no tasks for today. Add a new task by entering "yoda diary nt"')
 
+def search():
+    """ 
+    search for a note
+    """ 
 
+    if os.path.isfile(TODAYS_NOTES_ENTRY_FILE_PATH):
+        with open(TODAYS_NOTES_ENTRY_FILE_PATH) as todays_notes_entry:
+            contents = yaml.load(todays_notes_entry)
+            click.echo('Input the string you want to search for') 
+            search_string = raw_input().strip()
+
+            found = 0
+            for entry in contents['entries']:
+                text = entry['text']
+                time = entry['time']
+                if search_string.lower() in text.lower():
+                    if found==0:
+                        click.echo('Following notes match your search:')
+                        click.echo("--------------")
+                        click.echo("  Time  | Note")
+                        click.echo("--------|-----")
+                    click.echo(time + "| " + text)
+                    found=1
+
+            if found==0:
+                click.echo('No notes found having the above substring')
+
+    else:
+        chalk.red(
+            'There are no notes for today. Add a new note by entering "yoda diary nn"')
+
```

Comment:
Sorry, this is not what I meant by searching the notes. I would like to give the user an option to search in the previous notes. There is no point in searching today's notes, right?